"""Реализуйте класс PositiveList, унаследовав его от класса list, для хранения положительных целых чисел.

Также реализуйте новое исключение NonPositiveError.

В классе PositiveList переопределите метод append(self, x) таким образом, чтобы при попытке добавить неположительное
целое число бросалось исключение NonPositiveError и число не добавлялось, а при попытке добавить положительное целое
число, число добавлялось бы как в стандартный list.

В данной задаче гарантируется, что в качестве аргумента x метода append всегда будет передаваться целое число.

Примечание:
Положительными считаются числа, строго больше нуля.
"""


########################################################################################################################

class NonPositiveError(Exception):  # определяем новый класс NonPositiveError, наследующийся от класса Exception
    pass  # в данном случае, класс NonPositiveError ничего не делает, поэтому просто используем оператор pass


class PositiveList(list):  # определяем новый класс PositiveList, наследующийся от класса list
    # определяем конструктор класса PositiveList, принимающий необязательный параметр - список items
    def __init__(self, items=None):
        if items is None:  # проверяем, передан ли список items, если нет, то создаем пустой список
            items = []
        super().__init__(items)  # вызываем конструктор базового класса list и передаем ему список items

    def append(self, value):  # переопределяем метод append в классе PositiveList
        if value > 0:  # если значение value больше нуля
            super().append(value)  # вызываем метод append базового класса list и добавляем value в список
        else:  # иначе, если значение value не больше нуля
            # вызываем исключение NonPositiveError с сообщением "Non-positive value"
            raise NonPositiveError("Non-positive value")


########################################################################################################################


pl = PositiveList([1, 2, 3])
pl.append(4)
print(pl)  # Output: [1, 2, 3, 4]

try:
    pl.append(-1)
except NonPositiveError as e:
    print(e)  # Output: Non-positive value
